packages/mermaid/src/diagrams/git/gitGraphParser.ts

```diff
@@ -0,0 +1,210 @@
+import type { GitGraph } from '@mermaid-js/parser';
+import { parse } from '@mermaid-js/parser';
+import type { ParserDefinition } from '../../diagram-api/types.js';
+import { log } from '../../logger.js';
+import { populateCommonDb } from '../common/populateCommonDb.js';
+import { db } from './gitGraphAst.js';
+import { commitType } from './gitGraphAst.js';
+import type {
+  CheckoutAst,
+  CherryPickingAst,
+  MergeAst,
+  CommitAst,
+  BranchAst,
+  GitGraphDBParseProvider,
+} from './gitGraphTypes.js';
+
+const populate = (ast: GitGraph, db: GitGraphDBParseProvider) => {
+  populateCommonDb(ast, db);
+  // @ts-ignore: this wont exist if the direction is not specified
+  if (ast.dir) {
+    // @ts-ignore: this wont exist if the direction is not specified
+    db.setDirection(ast.dir);
+  }
+  for (const statement of ast.statements) {
+    parseStatement(statement, db);
+  }
+};
+
+const parseStatement = (statement: any, db: GitGraphDBParseProvider) => {
+  const parsers: Record<string, (stmt: any) => void> = {
+    Commit: (stmt) => db.commit(...parseCommit(stmt)),
+    Branch: (stmt) => db.branch(...parseBranch(stmt)),
+    Merge: (stmt) => db.merge(...parseMerge(stmt)),
+    Checkout: (stmt) => db.checkout(parseCheckout(stmt)),
+    CherryPicking: (stmt) => db.cherryPick(...parseCherryPicking(stmt)),
+  };
+
+  const parser = parsers[statement.$type];
+  if (parser) {
+    parser(statement);
+  } else {
+    log.error(`Unknown statement type: ${statement.$type}`);
+  }
+};
+
+const parseCommit = (commit: CommitAst): [string, string, number, string[] | undefined] => {
+  const id = commit.id;
+  const message = commit.message ?? '';
+  const type = commit.type !== undefined ? commitType[commit.type] : commitType.NORMAL;
+  const tags = commit.tags ?? undefined;
+
+  return [message, id, type, tags];
```

Comment:
These should be objects, not arrays, so it's safer. 
The db function can also be changed. 

Same with the rest

Reply:
Good idea